nodl_python/nodl/_parsing/_v1/_parsing.py

```diff
@@ -42,17 +55,20 @@ def _parse_parameter(element: etree._Element) -> Parameter:
 
 def _parse_service(element: etree._Element) -> Service:
     """Parse a NoDL service from an xml element."""
-    name = element.get('name')
-    service_type = element.get('type')
+    args = {}
+    args['name'] = element.get('name')
+    args['service_type'] = element.get('type')
 
-    server = get_bool_attribute(element, 'server')
-    client = get_bool_attribute(element, 'client')
-    if not (server or client):
+    args['server'] = get_bool_attribute(element, 'server')
+    args['client'] = get_bool_attribute(element, 'client')
+    if not (args['server'] or args['client']):
         raise errors.AmbiguousServiceInterfaceError(element)
 
-    policy = _parse_qos(element.find('qos'))
+    qos_element = element.find('qos_profile')
```

Comment:
I understand the change required for actions (where each action component has its own qos), but here we're just changing the name `qos` to `qos_profile`, is that right? This doesn't seem worthwhile, can you fill me in on the context? This comment applies to the same change for topics.

Reply:
Since we're changing Action and using the actual property name from rclpy, I figured uniformity and matching rclpy were a good idea to add in when there was an API breakage already happening.

Reply:
We don't _have_ to take the names verbatim from rclpy, we just need to offer the functionality. We could use `goal_service_qos` just as easily as `goal_service_qos_profile`. Are you sure you like `_profile` for all of them? Given that this library has been released with no immediate users other than ourselves I'm okay with the breakage, but let's make sure we're happy with the names this time around.

Reply:
This is a matter of opinion but I prefer matching exactly the fields we're representing, but we do have a discrepency in the case of Actions anyway as they specify either `pub` or `sub` profiles for server or client respectively.

I vote for `qos_profile` as it was an oversight rather than a choice to use `qos` originally on my part.

Reply:
I think `qos_profile` is a bit more verbose than necessary, but I see the benefit of matching rclpy here. I don't see the benefit of rewriting this particular logic to use a dict though, can you just change the key here like you did for topics?

Reply:
done in 354a1b6